### PR TITLE
Update Program.cs

### PR DESCRIPTION
**Pull request by CCianfloneDev**
Added exception handling, entering wrong format no longer abends program.

### CODE REVIEW

**Comment by CCianfloneDev** [comment]
Related to issue #13

File: ParkingGarageRentals/Program.cs [hunk 1]
```diff
@@ -32,12 +32,22 @@ static void Main(string[] args)
...
 
-                passNumberSelected = Convert.ToInt32(Console.ReadLine());
+                try
+                {
+                    passNumberSelected = Convert.ToInt32(Console.ReadLine());
+                }
+                catch (FormatException)
+                {
+                    Console.WriteLine("Must enter a valid number.");
+                }
```
Comment (DZ-FSDev) [inline ParkingGarageRentals/Program.cs]: - [x] do-while can be used here to ensure the user enters a valid number before continuing to logic below
- [x] a sentinel value can also be used to exit the menu

File: ParkingGarageRentals/Program.cs [hunk 2]
```diff
@@ -32,12 +32,22 @@ static void Main(string[] args)
...
 
                 switch (passNumberSelected)
                 {
-                    case int n when (n > 4 || n < 1):
+                    case 9:
+                        break;
+                    case int n when (n < 1 || n > 4):
                         Console.WriteLine("Please enter a valid choice between 1 - 4");
+                        Console.ReadKey();
```
Comment (DZ-FSDev) [inline ParkingGarageRentals/Program.cs]: - [x] this logic should be moved up so that this switch deals only with valid passes

**Review by DZ-FSDev — CHANGES_REQUESTED** [review]
**CR for #16 **

File: ParkingGarageRentals/Program.cs [hunk 3]
```diff
@@ -31,23 +31,37 @@ static void Main(string[] args)
...
+
 
                 switch (passNumberSelected)
                 {
                     case 9:
-                        break;
-                    case int n when (n < 1 || n > 4):
-                        Console.WriteLine("Please enter a valid choice between 1 - 4");
-                        Console.ReadKey();
+                        isNotSentinelValue = false;
```
Comment (DZ-FSDev) [inline ParkingGarageRentals/Program.cs]: - [x] The sentinel value is dealt with in the loop above; case 9 can be removed

File: ParkingGarageRentals/Program.cs [hunk 4]
```diff
@@ -67,7 +81,7 @@ static void Main(string[] args)
                         break;
                 }
 
-            } while (passNumberSelected != SentinelValue);
+            } while (isNotSentinelValue);
```
Comment (DZ-FSDev) [inline ParkingGarageRentals/Program.cs:78]: Can you clarify what this outer loop does? The inner loop deals with the sentinel.

File: ParkingGarageRentals/Program.cs [hunk 5]
```diff
@@ -31,23 +31,37 @@ static void Main(string[] args)
...
-                }
-                catch (FormatException)
+                do
                 {
-                    Console.WriteLine("Must enter a valid number.");
-                }
+                    if (passNumberSelected == 9)
+                    {
+                        isNotSentinelValue = false;
+                    }
```
Comment (DZ-FSDev) [inline ParkingGarageRentals/Program.cs]: - [x] Here's a place we can break out of the loop after setting `isNotSentinelValue = false;`
- [x] Wrap lines 86-eom if-statement on `isNotSentinelValue` (if user presses 9, it should just exit this menu; no more processing in this method needs to occur)

**Review by DZ-FSDev — CHANGES_REQUESTED** [review]
**CR for #16**

File: ParkingGarageRentals/Program.cs (hunk 4 above)
Comment (CCianfloneDev) [reply ParkingGarageRentals/Program.cs:78]: I've now removed inner loop.

**Review by DZ-FSDev — APPROVED** [review]
Looks good!

**Pull request merged**